runtime/src/accounts_index.rs

```diff
@@ -1442,17 +1443,21 @@ impl<
                 let mut duplicate_keys = Vec::with_capacity(items.len() / 10);
                 let mut w_account_maps = self.account_maps[pubkey_bin].write().unwrap();
                 let mut insert_time = Measure::start("insert_into_primary_index");
-                items.into_iter().for_each(|(pubkey, new_item)| {
-                    let already_exists = self.insert_new_entry_if_missing_with_lock(
-                        pubkey,
-                        &mut w_account_maps,
-                        new_item,
-                    );
-                    if let Some((mut w_account_entry, account_info, pubkey)) = already_exists {
-                        w_account_entry.update(slot, account_info, &mut _reclaims);
-                        duplicate_keys.push(pubkey);
-                    }
-                });
+                items
+                    .into_iter()
+                    .for_each(|(pubkey, new_item, is_zero_lamport)| {
+                        let already_exists = self.insert_new_entry_if_missing_with_lock(
+                            pubkey,
+                            &mut w_account_maps,
+                            new_item,
+                        );
+                        if let Some((mut w_account_entry, account_info, pubkey)) = already_exists {
+                            w_account_entry.update(slot, account_info, &mut _reclaims);
+                            duplicate_keys.push(pubkey);
+                        } else if is_zero_lamport {
+                            duplicate_keys.push(pubkey);
```

Comment:
nit: how about renaming `duplicate_keys` to `dirty_pubkeys`:

- _duplicate_ is now misnomer as it technically can contain not-duplicate zero-lamport keys.
- this is what is called at the sole call site (= `generate_index_for_slot`)